src/internal_events/kafka.rs

```diff
@@ -149,6 +149,12 @@ impl InternalEvent for KafkaStatisticsReceived<'_> {
             "kafka_consumed_messages_bytes_total",
             self.statistics.rxmsg_bytes as u64
         );
+
+        for (topic_id, topic) in &self.statistics.topics {
+            for (partition_id, partition) in &topic.partitions {
+                gauge!("kafka_consumer_lag", partition.consumer_lag as f64, "topic_id" => topic_id.clone(), "partition_id" => partition_id.to_string());
+            }
+        }
```

Comment:
This reads like it could cause quite a large number of new metrics. Is this something we should be making optional, or is it fine here because the number is relatively bounded?

Reply:
My guess that it should be relatively bounded. I don't think that it'll affect metrics number a lot. So I suggest to leave it as is, without an optional switch.

If users will complain about the metrics number or something like that - we will decide, what should be done here. And yes, one of the ways is to add an additional flag (or something like that).

Reply:
Normally you can filter the unused internal metrics with a filter transform. I think this is a really nice to have metric if you use kafka as source, as you don't need to install an extra kafka lag exporter

Reply:
Internal metrics can be filtered before exporting them, but they will result in a higher number of metrics needing to be tracked internally.

I agree that it'd be good to let users opt into this. We've already had a few issues resulting from tagging metrics with high cardinality tags which prompted #15426 . We likely need to come up with a more generic way to opt into these extra metrics but for now we could just use a simple boolean flag on the source config.

Reply:
@jszwedko so what is the proposed way to move this PR further? Adding some specific field like `enable_lag_metrics: bool` to the source/sink config? In this case, we need to update both Kafka source and sink configurations, not just a source configuration.

Reply:
Hey! Yeah, I think something like that would be fine. I'd suggest `metrics.topic_lag_metric: true|false` to group together any metric specific options like this. Does this metric apply to the sink? It seems to only apply to consumers (aka the source).

Reply:
> Does this metric apply to the sink? It seems to only apply to consumers (aka the source).

I guess it should. Because the same code is reused in the Kafka sink too.

Reply:
I agree on making this opt-in given we're adding `T x P` new metrics, without any knowledge of what those numbers look like for typical users.

Reply:
Fixed.